plugins/org.obeonetwork.m2doc.sirius/src/org/obeonetwork/m2doc/sirius/providers/tables/AbstractSiriusTableProvider.java

```diff
@@ -1,54 +1,79 @@
-            result.add(new DMTable(table));
-        }
-        return result;
-    }
-
-    /**
-     * This default implementation does nothing.
-     * 
-     * @return an empty list, never <code>null</code> but always empty.
-     */
-    @Override
-    public List<ProviderValidationMessage> validate(Map<String, Object> options) {
-        return Collections.emptyList();
-    }
-
-}
+/*******************************************************************************
+ * Copyright (c) 2016 Obeo. 
+ *    All rights reserved. This program and the accompanying materials
+ *    are made available under the terms of the Eclipse Public License v1.0
+ *    which accompanies this distribution, and is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ *     
+ *     Contributors:
+ *         Obeo - initial API and implementation
+ *******************************************************************************/
+package org.obeonetwork.m2doc.sirius.providers.tables;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.emf.transaction.RecordingCommand;
+import org.eclipse.sirius.business.api.session.Session;
+import org.eclipse.sirius.table.metamodel.table.DTable;
+import org.eclipse.sirius.viewpoint.DRepresentation;
+import org.obeonetwork.m2doc.provider.AbstractTableProvider;
+import org.obeonetwork.m2doc.provider.ProviderValidationMessage;
+
+/**
+ * Abstract super-implementation of all Sirius table providers.
+ * 
+ * @author ldelaigue
+ */
+public abstract class AbstractSiriusTableProvider extends AbstractTableProvider {
+
+    /**
+     * Extract the given tables as {@link MTable}s.
+     * 
+     * @param tables
+     *            The sirius tables to extract
+     * @return A list, never <code>null</code> but possibly empty, of tables in the M2Doc format.
+     */
+    protected List<MTable> extractTables(List<DTable> tables) {
+        List<MTable> result = new ArrayList<MTable>();
+        for (DTable table : tables) {
+            result.add(new DMTable(table));
+        }
+        return result;
+    }
+
+    /**
+     * This default implementation does nothing.
+     * 
+     * @return an empty list, never <code>null</code> but always empty.
+     */
+    @Override
+    public List<ProviderValidationMessage> validate(Map<String, Object> options) {
+        return Collections.emptyList();
+    }
+
+    /**
+     * Refresh representation before produce table in documentation.
+     * 
+     * @param table
+     *            sirius table representation
+     * @param session
+     *            session
+     * @param refreshTableFlag
+     *            refresh Table Flag
+     */
+    protected void refreshTable(final DRepresentation table, Session session, Boolean refreshTableFlag) {
+        if (refreshTableFlag) {
+            RecordingCommand recordCommand = new RecordingCommand(session.getTransactionalEditingDomain()) {

```

Comment:
This means that generating documents on an existing models will create undoable changes to the model.
I also imagine that if we run a generation on a clean session (one that doesn't need to be saved), then the end user would be a bit surprised that the session becomes dirty.
I think it may be hard to understand for end users so at least it should be documented.